pkg/kubelet/dockertools/docker.go

```diff
@@ -153,6 +154,14 @@ func filterHTTPError(err error, image string) error {
 
 // Check if the inspected image matches what we are looking for
 func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
+	// NB: the code below this check checks for validity of `image` as a pull spec.
+	// However, it can be useful to also inspect images by ID (config digest, e.g.
```

Comment:
The purpose of this function is to determine if the `image` parameter matches the `inspected` image, either by tag or digest. Because there are 3 ways a user can reference an image (tag, digest, id aka config.digest), we have to make sure that the inspected image does not match id/config.digest.

Because `inspected.ID` is the value of `config.digest`, the only way for it to equal `image` is if the user specified the `config.digest` for `containers[i].image` in the pod spec. But you can't pull by `config.digest`, which means that the only way you can successfully run a container whose image is `config.digest` is if that image already exists on the node. So you would have to:

1. Run a pod whose image is e.g. `foo/bar:latest`
1. Find out what that image's `config.digest` is
1. Run another pod, specifying the `config.digest` value for the image, **and** make sure the pod is scheduled to the same node from step 1